trufflesuite/truffle
# Further work on @truffle/plugins

This builds on the initial work done by @gnidan to pull out the plugin functionality into a separate package.
Contains these changes from his original code:

- Further split up `plugin.ts` into `Plugin.ts`, `Plugins.ts`, `utils.ts` and `interfaces.ts`
- Add `findPluginsForCommand()` function
- Fix `pluginPath` being used instead of `modulePath` in `Plugin:loadModule()`
- Add tests to `@truffle/plugins` package
- Add `@truffle/plugins` as dependency to `@truffle/core`
- Update `run.js` to use the new interface
- Update `run.js` tests to fit the new interface
- Update tests to check for specific error strings
- Update tests to assert that throwing tests actually throw
- Remove `plugin.js` tests since this functionality moved to `@truffle/plugins`
- Add `checkPluginConfig.js` tests (this is a subset of the old `plugin.js` tests)
- Do not catch loadModule() errors so misconfigurations get caught, rather than ignored
